### PR TITLE
model-controller-for-notification

### PR DESCRIPTION
**Pull request by jethoo**
### What this PR does (required):
**Post request to '/notification' , would create notifications for a user**
- body.to = user to whom the notification is being sent
- body.notification = notification message we want to convey

**Get request to '/notification', would give all the notifications sent to the logged-in user**

**Delete request to '/notification/delete' , would delete all the notifications for the logged-in user**

### Screenshots / Videos (required):
- Post at least one screenshot to show the feature in action
Screen Shot 2021-07-21 at 12 10 10

### Any information needed to test this feature (required):
- Post what steps to take to be able to test this feature (eg. "sign up as a new user and upload a profile picture from the navbar")

### Any issues with the current functionality (optional):
- List any problems where you were not able to complete all tasks on ticket
- List any problems this PR may cause for the project or other active PRs
- Post a screenshot/video of the problem, along with detailed console outputs where applicable

### CODE REVIEW

File: server/models/Notification.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  from: {
+    type: mongoose.Schema.Types.ObjectId,
+    ref: "User",
+    required: true
+  },
+  notification: {
+      type: [String],
+      required: true
+  }
+});
```
Comment (rajivtitus) [inline server/models/Notification.js]: Maybe also add timestamps?

Ref: https://masteringjs.io/tutorials/mongoose/timestamps

File: server/controllers/notification.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    const previousNotification = await Notification.findOne({
+      to: body.to,
+      from: req.user.id,
+    });
+
+    if (previousNotification) {
+      previousNotification.notification =
+        previousNotification.notification.concat(body.notification);
+      await previousNotification.save();
+      res.status(200).json(previousNotification);
```
Comment (rajivtitus) [inline server/controllers/notification.js:17]: Please use status code for resource creation

Ref: https://www.restapitutorial.com/httpstatuscodes.html

File: server/controllers/notification.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+    res.status(200).json({
+      allNotifications,
+    });
+  } catch (err) {
+    res.status(500).json(err);
+  }
+});
+
+exports.deleteAllNotifications = asyncHandler(async (req, res) => {
```
Comment (rajivtitus) [inline server/controllers/notification.js]: Wouldn't it be better to have the option to delete individual notifications?

**Review by rajivtitus — COMMENTED** [review]
Looks good for a start!

A couple of comments, please have a look

File: server/models/Notification.js (hunk 1 above)
Comment (jeewanbhusal-1) [reply server/models/Notification.js]: added timestamps.

File: server/controllers/notification.js (hunk 3 above)
Comment (jeewanbhusal-1) [reply server/controllers/notification.js]: I think so :). 
In the second commit, modified the controller to be able to delete individual notifications

**Review by rajivtitus — APPROVED** [review]
LGTM!

**Pull request merged**